src/utils/intrusive_ptr_test.cc

```diff
@@ -155,6 +155,12 @@ TEST(IntrusivePtrTest, FactoryConstructionWorks) {
   EXPECT_EQ(another_ptr->count(), 11);
   EXPECT_EQ(another_ptr->id(), "AnotherValue");
   EXPECT_NE(ptr.get(), another_ptr.get());
+  // Note that another_ptr going out of scope would only cause the count to be
```

Comment:
This test was just to check that non-trivial constructors work. I think it was a bad idea to use `count` in the constructor.  Instead of doing this, can you just move `initial_count` from the constructor and change/remove this part of the test completely? I don't think we are using `initial_count` for tests in any way.

Reply:
Sounds good, I think that's much better as well.

Reply:
BTW, I meant remove `initial_count` from the constructor completely. I just realized that it typed it as "move `initial_count`" from constructor. :)

Reply:
No prob, that is what I have done :)